lib/models/LifecycleConfiguration.ts

```diff
@@ -917,11 +937,20 @@ export default class LifecycleConfiguration {
             'NoncurrentVersionTransition',
             'Transition',
         ];
-        validActions.forEach(a => {
-            if (rule[a]) {
-                actionsObj.actions.push({ actionName: `${a}` });
+        for (const action of validActions) {
+            if (!rule[action]) {
+                continue;
             }
-        });
+
+            const isRuleSupported = this._supportedLifecycleRules.includes(action);
+            if (!isRuleSupported) {
+                const msg = `${action} lifecycle action not implemented`;
+                const error = errorInstances.NotImplemented.customizeDescription(msg);
+                return { ...actionsObj, error };
+            }
+
+            actionsObj.actions.push({ actionName: `${action}` });
```

Comment:
Do we use `${...}` to copy the string? I don't think it's needed since we don't modify the contents of `action`, and strings are immutable anyways.